charts/application/templates/k8s-deployment.yaml

```diff
@@ -64,10 +64,16 @@ spec:
             - secretRef:
                 name: {{ include "secret" . }}
             {{- end }}
-          {{- if .Values.serviceAccount.secretName }}
+          {{- if or .Values.serviceAccount.secretName .Values.additionalEnv }}
           env:
+            {{- if .Values.serviceAccount.secretName }}
             - name: "GOOGLE_APPLICATION_CREDENTIALS"
               value: "{{- .Values.serviceAccount.mountPath -}}/key.json"
+            {{- end }}
+            {{- range .Values.additionalEnv }}
```

Comment:
redundant to 
```
          envFrom:
            - configMapRef:
                name: {{ include "configmap" . }}
            {{- range $c := .Values.configEnvFrom }}
```
but there are often multiple options, and your's is convenient.